lib/ResourceTokenizer.js

```diff
@@ -241,16 +241,17 @@ export class ResourceTokenizer {
       let toRemove = count;
       const newList = [];
       for(const e of list) {
+        const newItem = {count: e.count, requested: e.requested};
         if(toRemove >= e.count) {
           // do not include entry
           toRemove -= e.count;
           continue;
         }
         if(toRemove > 0) {
-          e.count -= toRemove;
+          newItem.count -= toRemove;
           toRemove = 0;
         }
-        newList.push(e);
+        newList.push(newItem);
```

Comment:
Yeah, it does seem like we need to ensure we don't modify the existing entry so that the atomic comparison check will work. This was working when the entire entry was removed because it wouldn't be modified; the new list just would omit it. Good find. I'll look more closely to double check this is the right fix shortly. I'm surprised we didn't have test coverage here either.

Regardless, I noticed a couple of things with this specific change:

1. `newItem` gets created and thrown away if the first conditional is true (it's wasteful).
2. `newItem` doesn't change if the second conditional isn't true (so if we don't need a copy here, it's waste).
3. The code would be more resilient if we shallow copied the entire `e`.

To remedy these things, we should do this instead (I couldn't easily make a github "suggestion" for the whole block, so here it is):

```js
      for(const e of list) {
        if(toRemove === 0) {
          // nothing to remove, keep existing entry
          newList.push(e);
          continue;
        }
        if(toRemove >= e.count) {
          // do not include entry at all, `toRemove` exceeds its count
          toRemove -= e.count;
          continue;
        }
        // include a copy of entry with a decremented count
        const newEntry = {...e};
        newEntry.count -= toRemove;
        newList.push(newEntry);
        toRemove = 0;
      }
```